### PR TITLE
Move Connexions to new structure

### PR DESCRIPTION
**Pull request by borisvida**

### CODE REVIEW

File: client/src/components/Connexions/dataWrappers/ConnexionWrapper.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import {compose} from 'redux'
+import {connect} from 'react-redux'
+import {withDataProviders} from 'data-provider'
+
+import {connexionDetailProvider} from '../../../dataProviders/connexionsDataProviders'
+
+const ConnexionWrapper = (WrappedComponent) => {
+  const wrapped = (props) => <WrappedComponent {...props} connexions={props.connexions} />
```
Comment (mpinter) [inline client/src/components/Connexions/dataWrappers/ConnexionWrapper.js]: doesn't this override the connexions you connect ?

File: client/src/components/Connexions/dataWrappers/ConnexionWrapper.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import React from 'react'
```
Comment (mpinter) [inline client/src/components/Connexions/dataWrappers/ConnexionWrapper.js]: :+1: I quite like the idea of separating these into files like this

File: client/src/components/Connexions/Connexions.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  }
+
+  render() {
+    const {entitySearch1, entitySearch2} = this.state
+    return (
+      <div className="container-fluid connexions">
+        <div className="row">
+          <div className="sidebar col-sm-5 col-md-4 col-lg-3">
+            <div id="myAffix" data-spy="affix">
+              <Search
```
[the remark this answers is not in the document]
Comment (borisvida) [reply client/src/components/Connexions/Connexions.js]: Yeah, good idea to have a single source of truth in url.

File: client/src/components/Connexions/dataWrappers/ConnexionWrapper.js (hunk 1 above)
Comment (borisvida) [reply client/src/components/Connexions/dataWrappers/ConnexionWrapper.js]: :man_facepalming: Yeah, doing connect's work...

File: client/src/components/Connections/scenes/Results/components/InfoLoader/InfoLoader.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+const InfoLoader = (props) => (
+  <div
+    className={
+      props.recursive
+        ? 'infoWrapper'
+        : 'infoWrapper col-md-offset-2 col-md-8 col-lg-offset-3 col-lg-6'
+    }
+  >
+    {<div>{props.data.entities[0].entity_name}</div>}
+    {/* <Info data={props.data} eid={props.eid} /> */}
```
Comment (mpinter) [inline client/src/components/Connections/scenes/Results/components/InfoLoader/InfoLoader.js]: I assume this is the componetn overlap with the part @MarekCacko was doing (just checking, in that case it's fine)

File: client/src/components/Connections/scenes/Results/Results.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
+// @flow
+import React from 'react'
+import ConnectionWrapper from '../../dataWrappers/ConnectionWrapper'
+import EntityWrapper from '../../dataWrappers/EntityWrapper'
+import InfoLoader from './components/InfoLoader/InfoLoader'
+
+const Results = (props) => (
+  <div>
+    {props.connections && props.connections.length > 0 ? (
```
Comment (mpinter) [inline client/src/components/Connections/scenes/Results/Results.js]: when you have components relying on something existing in state, you can leverage data-providers, which do not render anything until they have

if you need to display some component if you have a part of state setup and display something else if not, you can also do it with recompose and branch - https://github.com/acdlite/recompose/blob/master/docs/API.md#branch

File: client/src/components/Connections/scenes/Results/components/InfoLoader/InfoLoader.scss [hunk 6]
```diff
@@ -0,0 +1,32 @@
+.resultLoading {
+  background-color: #ffffff;
```
Comment (mpinter) [inline client/src/components/Connections/scenes/Results/components/InfoLoader/InfoLoader.scss]: you get the idea by now :)

File: client/src/components/Connections/scenes/Results/components/InfoLoader/InfoLoader.scss [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+  width: 400px;
+  display: block;
+  margin-left: auto;
+  margin-right: auto;
+  margin-bottom: 30px;
```
Comment (mpinter) [inline client/src/components/Connections/scenes/Results/components/InfoLoader/InfoLoader.scss]: it's also good to use rem on margins/paddings where it makes sense, as it scales better with the rest of the page, but I don't know the exact good practices here so keeping in px might be ok

File: client/src/dataProviders/connectionsDataProviders.js [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+const dispatchEntitiesData = (query: string) => (ref: string, data: any, dispatch: Dispatch) => {
+  dispatch(
+    receiveData(['connections', 'entities'], {id: query, eids: data.map(({eid}) => eid)}, ref)
+  )
+}
+
+const dispatchConnectionData = (eid1: string, eid2: string) => (
+  ref: string,
+  data: any,
```
Comment (mpinter) [inline client/src/dataProviders/connectionsDataProviders.js]: we can probably do better than any

File: client/src/dataProviders/connectionsDataProviders.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+  dispatch: Dispatch
+) => {
+  dispatch(receiveData(['connections', 'detail'], {id: `${eid1}-${eid2}`, ids: data}, ref))
+}
+
+const dispatchEntityDetailData = (eid: string) => (ref: string, data: any, dispatch: Dispatch) => {
```
Comment (mpinter) [inline client/src/dataProviders/connectionsDataProviders.js]: any again

File: client/src/dataProviders/connectionsDataProviders.js [hunk 10]
```diff
@@ -0,0 +1,80 @@
+// @flow
+import type {Dispatch} from '../types/reduxTypes'
+import {dispatchReceivedData} from './dataProvidersUtils'
+import {receiveData} from '../actions/sharedActions'
+
+const dispatchEntitiesData = (query: string) => (ref: string, data: any, dispatch: Dispatch) => {
```
Comment (mpinter) [inline client/src/dataProviders/connectionsDataProviders.js]: any are you ok ? are you ok any ? 🕺

File: client/src/state/index.js [hunk 11]
```diff
@@ -65,15 +65,32 @@ export type MapOptions = {
...
+  id: string,
+}
+
+export type Connections = {
+  entities: {[string]: Entity},
+  detail: {[string]: {ids: string[]}},
+  entityDetails: {
+    [string]: {
+      name: string,
+      data: any, //TODO: TBD
```
Comment (mpinter) [inline client/src/state/index.js]: oh, I see now - ok, disregard the Annies if they are related to this

File: client/src/components/Connections/Connections.scss [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+  height: 100%;
+}
+
+.statuses {
+  position: absolute;
+  top: 10px;
+  left: 0;
+  width: 100%;
+  text-align: center;
+  color: #859fb4;
```
[the remark this answers is not in the document]
Comment (borisvida) [reply client/src/components/Connections/Connections.scss]: Yup.

File: client/src/components/Connections/Connections.scss [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+  font-size: 15px; /* Approximation due to font substitution */
```
[the remark this answers is not in the document]
Comment (borisvida) [reply client/src/components/Connections/Connections.scss]: Changed to rem.

File: client/src/components/Connections/Connections.scss [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+.fbfooter .col-sm-10 {
+  padding-left: 0;
+}
+
+.fbIframe {
+  border: none;
+  overflow: hidden;
+}
+
+#myAffix {
```
[the remark this answers is not in the document]
Comment (borisvida) [reply client/src/components/Connections/Connections.scss]: Just copied the old css. Fixed.

File: client/src/components/Connections/Connections.js [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+import Search from './components/Search/Search'
+import Statuses from './components/Statuses/Statuses'
+import Results from './scenes/Results/Results'
+import './Connections.css'
+
+const Connections = () => (
+  <div className="container-fluid connections">
+    <div className="row">
+      <div className="sidebar col-sm-5 col-md-4 col-lg-3">
+        <div id="myAffix" data-spy="affix">
```
[the remark this answers is not in the document]
Comment (borisvida) [reply client/src/components/Connections/Connections.js]: Removed.

File: client/src/components/Connections/components/Search/Search.js [hunk 16]
```diff
@@ -0,0 +1,107 @@
+// @flow
+import React, {PureComponent} from 'react'
+import {withRouter} from 'react-router-dom'
+import type {RouterHistory} from 'react-router'
+
+import withEntitySearch from '../../dataWrappers/EntitySearchWrapper'
+import './Search.css'
+
+type Props = {|
+  entitySearch1: string,
```
[the remark this answers is not in the document]
Comment (borisvida) [reply client/src/components/Connections/components/Search/Search.js]: Renamed that object to `SearchedEntity`.

File: client/src/components/Connections/components/Statuses/Statuses.js [hunk 17]
```diff
@@ -0,0 +1,79 @@
...
+        {entity2 && (
+          <span id="search-status2" className="searchStatus">
+            {this.translateZaznam(entity2.eids.length, () =>
+              this.setState({showAlternatives2: true})
+            )}{' '}
+            pre &quot;{entity2.id}&quot;.&nbsp;
+          </span>
+        )}
+        {showAlternatives1 &&
+          entity1 &&
```
[the remark this answers is not in the document]
Comment (borisvida) [reply client/src/components/Connections/components/Statuses/Statuses.js]: Solved on call, moved the checks higher.

File: client/src/components/Connections/components/Statuses/Alternative.js [hunk 18]
```diff
@@ -0,0 +1,16 @@
+// @flow
```
[the remark this answers is not in the document]
Comment (borisvida) [reply client/src/components/Connections/components/Statuses/Alternative.js]: Nope. The app finds a set of entities for your entered string. It takes the first and looks for connections for it. `Alternatives` are the other found entities, if the first isn't the one you meant.

File: client/src/components/Connections/components/Statuses/Statuses.js [hunk 19]
```diff
@@ -0,0 +1,79 @@
...
+    }
+    return <span>Nájdených {button} záznamov</span>
+  }
+
+  render() {
+    const {entity1, entity2, connections} = this.props
+    const {showAlternatives1, showAlternatives2} = this.state
+    return (
+      <div className="statuses">
+        {connections &&
```
[the remark this answers is not in the document]
Comment (borisvida) [reply client/src/components/Connections/components/Statuses/Statuses.js]: Removed.

File: client/src/components/Connections/scenes/Results/components/InfoLoader/InfoLoader.js (hunk 4 above)
Comment (borisvida) [reply client/src/components/Connections/scenes/Results/components/InfoLoader/InfoLoader.js]: Yup, I hope so :slightly_smiling_face: @MarekCacko can you please confirm?

Comment (MarekCacko) [reply client/src/components/Connections/scenes/Results/components/InfoLoader/InfoLoader.js]: Yes, everything what you had in Info directory is copy of components from my code. Even this InfoLoader.js was in my code (now it has a new name)

File: client/src/components/Connections/scenes/Results/Results.js (hunk 5 above)
Comment (borisvida) [reply client/src/components/Connections/scenes/Results/Results.js]: Rewritten.

File: client/src/dataProviders/connectionsDataProviders.js (hunk 8 above)
Comment (borisvida) [reply client/src/dataProviders/connectionsDataProviders.js]: TBD :slightly_smiling_face:

File: client/src/dataProviders/connectionsDataProviders.js (hunk 9 above)
Comment (borisvida) [reply client/src/dataProviders/connectionsDataProviders.js]: TBD :slightly_smiling_face:

File: client/src/dataProviders/connectionsDataProviders.js (hunk 10 above)
Comment (borisvida) [reply client/src/dataProviders/connectionsDataProviders.js]: TBD :slightly_smiling_face:

File: client/src/state/index.js (hunk 11 above)
Comment (borisvida) [reply client/src/state/index.js]: Yup, they are :slightly_smiling_face:

File: client/src/components/Connections/scenes/Results/components/InfoLoader/InfoLoader.scss (hunk 7 above)
Comment (borisvida) [reply client/src/components/Connections/scenes/Results/components/InfoLoader/InfoLoader.scss]: I think the whole design and UX has to be reworked, so I don't want to dig into too much polishing.

File: client/src/components/Connections/Connections.scss [hunk 20]
```diff
@@ -0,0 +1,75 @@
...
+  flex-direction: column;
+  align-items: center;
+  background-color: #ffffff;
+  padding-right: 15px;
+  padding-left: 15px;
+}
+
+@media(max-width: 768px){
+    #myAffix{
+        position: relative !important;
```
[the remark this answers is not in the document]
Comment (borisvida) [reply client/src/components/Connections/Connections.scss]: Removed.

**Pull request merged**